dev/statement_serializer.h

```diff
@@ -45,81 +44,222 @@ namespace sqlite_orm {
+                                      const std::string& alias) {
+            constexpr char quoteChar = '"';
+
             if(!qualifier.empty()) {
-                quoted = quote_identifier(qualifier) + "." + move(quoted);
+                ss << quoteChar << sql_escape(qualifier, quoteChar) << std::string{quoteChar, '.'};
             }
+            ss << quoteChar << sql_escape(identifier, quoteChar) << quoteChar;
             if(!alias.empty()) {
-                quoted += " " + quote_identifier(alias);
+                ss << std::string{' ', quoteChar} << sql_escape(alias, quoteChar) << quoteChar;
             }
-            return quoted;
         }
 
-        inline std::string serialize_identifier(const std::string& identifier, const std::string& alias) {
-            std::string quoted = quote_identifier(identifier);
-            if(!alias.empty()) {
-                quoted += " " + quote_identifier(alias);
-            }
-            return quoted;
+        inline void stream_identifier(std::ostream& ss, const std::string& identifier, const std::string& alias) {
+            return stream_identifier(ss, std::string{}, identifier, alias);
         }
 
-        inline std::string serialize_identifier(const std::string& identifier) {
-            return quote_identifier(identifier);
+        inline void stream_identifier(std::ostream& ss, const std::string& identifier) {
+            return stream_identifier(ss, std::string{}, identifier, std::string{});
         }
 
-        template<typename T, size_t... Is>
-        std::string serialize_identifier(const T& tpl, std::index_sequence<Is...>) {
-            return serialize_identifier(std::get<Is>(tpl)...);
+        template<typename Tpl, size_t... Is>
+        void stream_identifier(std::ostream& ss, const Tpl& tpl, std::index_sequence<Is...>) {
+            return stream_identifier(ss, std::get<Is>(tpl)...);
         }
-        template<typename T>
-        std::string serialize_identifier(const T& tpl) {
-            return serialize_identifier(tpl, std::make_index_sequence<std::tuple_size<T>::value>{});
+
+        template<typename Tpl>
+        void stream_identifier(std::ostream& ss, const Tpl& tpl) {
+            return stream_identifier(ss, tpl, std::make_index_sequence<std::tuple_size<Tpl>::value>{});
         }
 
-        template<class It>
-        std::string serialize_identifiers(It first, It end) {
-            return std::accumulate(first, end, std::string{}, [](std::string a, const auto& tpl) {
-                constexpr const char* sep[] = {", ", ""};
-                return move(a) + sep[a.empty()] + serialize_identifier(tpl);
+        enum class stream_as {
+            conditions_tuple,
+            actions_tuple,
+            expressions_tuple,
+            dynamic_expressions,
+            serialized,
+            identifier,
+            identifiers,
+            values_placeholders,
+        };
+
+        template<stream_as mode>
+        struct streaming_ {
```

Comment:
You are asking tough questions that force me to verifying my own subtle choices 🤔 
Did you see the constexpr instantiations below, e.g. `streaming_expressions_tuple`? The underscore fits with the continuation in whatever words you append to it - `streaming_...`
Though I realize it might escape usual naming conventions. Do you have a better suggestion?

Reply:
yes: `streaming`. Underscore in the end is used in `sqlite_orm` only if omitting underscore will make a reserved C++ keyword. This is how it work in `boost` also